### PR TITLE
Add CLI to filecoindot

### PR DESCRIPTION
**Pull request by ivanceras**
## Changes

- Add a `fn generic_verify` in `filecoindot_proofs/src/verify.rs` module
- Add `#[derive(this::Error)]` anotation for `filecoindot_proofs::Error` to easily convert and forward error to `fn main`.

## Tests

```
cargo test
cargo test --release

```

## Issues

- #2 
- #109

### CODE REVIEW

File: filecoindot_proofs/src/verify.rs [hunk 1]
```diff
@@ -44,3 +45,10 @@ impl ProofVerify {
         Self::traverse_and_match::<N>(proof, index - 1, target_cid)
     }
 }
+
+pub(crate) type NodeType =
```
Comment (ivanceras) [inline filecoindot_proofs/src/verify.rs:49]: Note: `NodeType` is copied from `forest_hamt_adaptor` test.

File: cli/tests/test_all.rs [hunk 2]
```diff
@@ -0,0 +1,624 @@
+use cid::Cid;
+use filecoindot_cli::{hex_encode_proof, NodeType};
+use filecoindot_proofs::deserialize_to_node;
+use filecoindot_proofs::traits::GetCid;
+use filecoindot_proofs::ProofVerify;
+use std::convert::TryFrom;
+
+#[test]
+fn verify_works() {
+    let p1 = vec![
```
Comment (ivanceras) [inline cli/tests/test_all.rs:10]: This is derived from print debugging the generated proof used in `forest_hamt_adaptor` test.

**Review by mattsse — APPROVED** [review]
lgtm

File: filecoindot_proofs/src/verify.rs (hunk 1 above)
Comment (ivanceras) [reply filecoindot_proofs/src/verify.rs:49]: @willeslau is it a good idea to make this`NodeType` be in the filecoindot_proofs? This is needed in order to expose a `generic_verify` method.

**Review by mattsse — APPROVED** [review]
awesome,
there are some missing license comments I believe,
otherwise lgtm,
merge wen ready

**Pull request merged**